mesh/activation.go

```diff
@@ -101,3 +127,106 @@ func BytesAsAtx(b []byte) (*ActivationTx, error) {
 	}
 	return &atx, nil
 }
+
+type ActiveSetProvider interface {
+	GetActiveSetSize(l LayerID) uint32
+}
+
+type MeshProvider interface {
+	GetLatestView() []BlockID
+	LatestLayerId() LayerID
+}
+
+type EpochProvider interface {
+	Epoch(l LayerID) EpochId
+}
+
+type Broadcaster interface {
+	Broadcast(channel string, data []byte) error
+}
```

Comment:
like i said im not sure this is a good idea
i think it should be in a separate package 
if the types are an issue maybe its time to pull them out of here as well